### PR TITLE
Check for null before checking for type IEnumerable.

### PR DESCRIPTION
**Pull request by josteink**
This closes https://github.com/nunit/nunit/issues/2486.

### CODE REVIEW

**Comment by CharliePoole** [comment]
Looks good to me.

**Comment by josteink** [comment]
I like that. I’ll make it so.

**Comment by jnm2** [comment]
:+1: `ConstraintUtils` was just off the top of my head, so if you find an existing helper class that seems related, we should probably use that.

**Comment by josteink** [comment]
I've updated the code in the places I've found which needed changing, and I've corrected the issues identified by the CI-build.

If you think there are more changes which needs to be made, please let me know. Otherwise I'll assume this is ready for a new review :)

File: src/NUnitFramework/framework/Internal/ConstraintUtil.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    internal class ConstraintUtil
+    {
+        /// <summary>
+        /// Require that the provided object is actually of the type required,
+        /// and that it is not null.
+        /// <param name="actual">The object to verify</param>
+        /// <param name="paramName">Name of the parameter as passed into the checking method.</param>
+        /// <typeparam name="T">The type to require</typeparam>  
+        /// <returns>A properly typed object, or throws. Never null.</returns>
+        /// </summary>
```
Comment (mikkelbu) [inline src/NUnitFramework/framework/Internal/ConstraintUtil.cs]: I usually put `param`, `typeparam`, and `returns` outside the `summary` element (I don't know if this makes a difference or not). Also it is a bit inconsisten whether the the sentence ends with a period.

File: src/NUnitFramework/framework/Internal/ConstraintUtil.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// ***********************************************************************
+// Copyright (c) 2015 Charlie Poole, Rob Prouse
```
[the remark this answers is not in the document]
Comment (josteink) [reply src/NUnitFramework/framework/Internal/ConstraintUtil.cs]: Fair enough. Will fix.

File: src/NUnitFramework/framework/Internal/ConstraintUtil.cs (hunk 1 above)
Comment (josteink) [reply src/NUnitFramework/framework/Internal/ConstraintUtil.cs]: I agree that makes sense. Will fix.

**Comment by josteink** [comment]
> Also, I really like the new method. Is there any reason why you did not use it in

Yes. I realized such a reusability concern could possibly arise in the future, but decided that this could be a good "base-commit" to land properly before making any changes other aspects of the framework.

There were also the case of poor tooling. I was fixing up the PR on my home machine running whatever version of mono ships with Fedora, and I couldn't get the whole build-chain running (I multiple causes of build-failure with the Linux-solution).

I therefore decided to be defensive about it, and not touch any more components than I felt I could guarantee the correctness of.

**Comment by josteink** [comment]
I've pushed a final commit which should address most of the concerns @mikkelbu had.

**Comment by ChrisMaddock** [comment]
image

@mikkelbu - do you not see "Dismiss review" at the bottom here? We should have the same access rights, I believe.

Merging - thanks @josteink! We'd be interested to learn about/fix your mono problems - if you'd fancy debugging and opening an issue? You can see the mono versions we currently test in CI in the .travis.yml file.

**Comment by ChrisMaddock** [comment]
(Also, @mikkelbu - can I leave it with you to open an issue re: the additional changes? Thanks!)

**Comment by jnm2** [comment]
Ooh, race condition! I was going to request a change but we can do it later, it's internal and minor.

**Comment by jnm2** [comment]
(`static` on the new class)

**Comment by ChrisMaddock** [comment]
Ah, sorry! I was happy it was internal, so didn't read further down the line! 🙈

**Comment by jnm2** [comment]
@ChrisMaddock That's not actually a bad strategy! :D

**Comment by mikkelbu** [comment]
@ChrisMaddock  I don't think I've the same rights. I went through all the open PRs and all I got was the following
image
But accordingly to the guide (https://help.github.com/articles/dismissing-a-pull-request-review/) "repository administrators or **people with write access** can dismiss a review." I should have enough?

Ps. I'll create the new issue

**Comment by ChrisMaddock** [comment]
@rprouse - could you investigate? Or @CharliePoole - if you're still an admin on the repo?

**Comment by CharliePoole** [comment]
Too late to investigate this one, as it is merged! In general, I can see "Dismiss review" on both my own reviews and those of others. However, Rob and I are owners of the organization so that doesn't prove much.

**Comment by ChrisMaddock** [comment]
Sorry - I meant could you investigate for any differences between mine and @mikkelbu's admin rights. AFAIK, they should be the same!

**Comment by ChrisMaddock** [comment]
To be clear, currently: I can dismiss reviews and @mikkelbu can't. I believe we should have the same permissions for this!

**Comment by CharliePoole** [comment]
I can't see any difference in GitHub. Two thoughts:

* @mikkelbu are you sure you were logged in?
* Could it be a browser issue? I use Firefox.

**Comment by mikkelbu** [comment]
@CharliePoole I'm quite sure that I'm logged in. Also the box about "Checks and Change requests" in the bottom is not visible if one is not logged in (as far as I can tell from testing now). I've tried Chrome, IE, and Edge on all the open PRs in this project I see the same "See review", but no "Dismiss review".

**Comment by CharliePoole** [comment]
As far as I can see, the only difference is that @ChrisMaddock is on the Core Team. But that doesn't grant any special rights as an admin or maintainer. @rprouse can you see anything?

**Pull request merged**